test/jdk/java/lang/Math/AbsTests.java

```diff
@@ -143,4 +181,55 @@ private static int testLongAbs(LongUnaryOperator absFunc,
             return 0;
         }
     }
+
+    // --------------------------------------------------------------------
+
+    private static int testFloatAbs() {
+        DoubleStream doubles = DoubleStream.of(FLOATING_POINT_VALUES);
+
+        final AtomicInteger errors = new AtomicInteger();
+        doubles.mapToObj(d -> (float)d).
+            forEach(f -> {errors.addAndGet(testFloatAbs(Math::abs, f));});
+
+        return errors.get();
+    }
+
+    private static int testFloatAbs(UnaryOperator<Float> absFunc, float f) {
+        float result = absFunc.apply(-f);
+        if (Float.isNaN(f)) {
+            return Float.floatToRawIntBits(result) !=
+                Float.floatToRawIntBits(f) ? 1 : 0;
+        } else if ((f >= 0 && result != f) || (f < 0 && result != -f)) {
```

Comment:
Please look at the Tests.java file in the jdk/test/java/lang/Math directory. One of the methods like

    public static int test(String testName, double input,
                           double result, double expected

will be helpful here; e.g.

Tests.test("Math.abs", f, Math.abs(-f), f)

I would also add combination for StrictMath.abs and negations:

Tests.test("Math.abs", f, Math.abs(f), f)
Tests.test("StrictMath.abs", -f, Math.abs(f), f)
Tests.test("StrictMath.abs", f, Math.abs(f), f)

Reply:
Modified.